### PR TITLE
Fix op-badge not getting the actual OP in forums

### LINKED ISSUE #6213: In the forums, "Highlight project creator in comments" calls the first poster of a page the OP.
### Description

When a forum topic has more than one page, it would be expected that the person marked as the OP is the OP. Instead, it's always the first poster on the page.

### Steps to reproduce

1. Go to any page of a topic where the first poster isn't the OP, like https://scratch.mit.edu/discuss/topic/614802/?page=42
2. Go to a post of that same user, like https://scratch.mit.edu/discuss/post/7315597/

### Demonstration(s) and additional context

_Actual OP_
Actual OP

_First post of a page marked as OP_
First post of a page marked as OP

### Environment

Browser: Edge 114
Scratch Addons version: 1.33.0
Operating system: Windows 11

### PR DESCRIPTION
**Pull request by mybearworld**
Resolves #6213

### Changes

Adds an additional pair of brackets (and also uses location.pathname instead of splitting the location.href because that's also error prone).

### Reason for changes

To fix a bug.

### Tests

Tested on Edge.

### CODE REVIEW

**Review by apple502j — APPROVED** [review]
how did https://eslint.org/docs/latest/rules/no-constant-binary-expression not catch this

File: addons/op-badge/forums.js [hunk 1]
```diff
@@ -2,11 +2,11 @@ export default async function ({ addon, console, msg }) {
   const posts = document.querySelectorAll(".blockpost");
   let opUsername;
 
-  if (new URLSearchParams(location.search).get("page") || 1 === 1) {
+  if ((new URLSearchParams(location.search).get("page") || 1) === 1) {
```
Comment (apple502j) [inline addons/op-badge/forums.js:5]:
```suggestion
  if ((new URLSearchParams(location.search).get("page") || "1") === "1") {
```

**Review by mxmou — APPROVED** [review]

**Comment by WorldLanguages** [comment]
Very funny bug right here.

**Comment by medlans** [comment]
LOL, this reminds me of this bug:
https://github.com/scratchfoundation/scratch-www/commit/c2cddbf63346cba004f105a2df0ddb30f049290f

**Review by WorldLanguages — APPROVED** [review]
Tested successfully in Chromium in this 3 URLs:

1) https://scratch.mit.edu/discuss/topic/444229/
2) https://scratch.mit.edu/discuss/topic/444229/?page=1
3) https://scratch.mit.edu/discuss/topic/444229/?page=2

**Pull request merged**